Lib/importlib/_bootstrap_external.py

```diff
@@ -960,7 +960,7 @@ class _NamespacePath:
 
     def __init__(self, name, path, path_finder):
         self._name = name
-        self._path = path
+        self._path = path.__fspath__() if hasattr(path, '__fspath__') else path
```

Comment:
This should be reverted as `_NamespacePath` is private and this affects what is valid for `__path__` which I'm not prepared to change.